### ISSUE TITLE
Detailed explanation about the training data

### ISSUE THREAD

**Issue by rose-jinyang** [lizhengwei1992/mobile_phone_human_matting#13]
Hello
How are you?
Thanks for contributing this project.
Could u provide a detailed explanation about the training data?
Thanks

**Comment by xiaochuan-cd** [comment]
My method for making dataset:

1、download from: https://github.com/aisegmentcn/matting_human_datasets
      The dataset name is matting_human_half.zip, about 15GB.

2、train.txt 

scripts for generting train,txt:

`import os
files = open('dataset/train.txt').read().split('\n')
for i in files:
    x, y = i.split()
    assert os.path.exists('dataset/'+x), 'dataset/'+x
    assert os.path.exists('dataset/'+y), 'dataset/'+y`

3、mask.
 _mask = cv2.imread('1803151818-00000003.png').astype(np.float32)
  _mask = (_mask!=0).astype(float)

You can and the second line to data,py, line no:119

**Comment by xiaochuan-cd** [comment]
My method for making the dataset:

1、Download from: https://github.com/aisegmentcn/matting_human_datasets
The dataset name is matting_human_half.zip, about 15GB.

2、Making train.txt
Every line like bellow: 
human_matting/clip_img/1803151818/clip_00000000/1803151818-00000003.jpg human_matting/matting/1803151818/matting_00000000/1803151818-00000003.png
Split tow full names with a space.

scripts for generating train.txt:

import os
train_txt = []
for root, dirs, files in os.walk("./clip_img", topdown=False):
    for name in files:
        line = 'human_matting' + os.path.join(root[1:], name).replace('\', '/')
        linet = line.replace('clip_img', 'matting').replace('.jpg', '.png').replace('clip_', 'matting_')
        if '._' not in line:
            train_txt.append(line+' '+linet)
with open('train.txt', 'w') as f:
    f.write('\n'.join(train_txt))

3、mask.
_mask = cv2.imread('1803151818-00000003.png').astype(np.float32)
_mask = (_mask!=0).astype(float)

You can insert the second line to data,py, line no:119

**Comment by xiaochuan-cd** [comment]
My method for making the dataset:

1、Download from: https://github.com/aisegmentcn/matting_human_datasets
The dataset name is matting_human_half.zip, about 15GB.

2、Making train.txt
Every line like bellow. Split tow relative path names with a space.

`human_matting/clip_img/1803151818/clip_00000000/1803151818-00000003.jpg human_matting/matting/1803151818/matting_00000000/1803151818-00000003.png`

3、mask.
```
_mask = cv2.imread('1803151818-00000003.png').astype(np.float32)
_mask = (_mask!=0).astype(float)
```
You can insert the second line to data.py, line no:119

**Comment by rose-jinyang** [comment]
Thanks for reply.
May I ask one more?
The model trained by u runs in the speed of about 100ms/frame on my PC.
I trained a new model with your project on AISegment dataset.
But this new model runs in the speed of about 450ms/frame.
The accuracies are almost same but not for speed.
What do u think the reason?
Thanks

**Comment by xiaochuan-cd** [comment]
> Thanks for reply.
> May I ask one more?
> The model trained by u runs in the speed of about 100ms/frame on my PC.
> I trained a new model with your project on AISegment dataset.
> But this new model runs in the speed of about 450ms/frame.
> The accuracies are almost same but not for speed.
> What do u think the reason?
> Thanks

I think the possible reson is that the provided pre-trained model trained with newer source code. The current repo source code is older, when run the pre-trained model , I noticed the warning:

**SourceChangeWarning: source code of class 'model.segnet.SegMattingNet' has changed.**

**Comment by xiaochuan-cd** [comment]
> Thanks for reply.
> May I ask one more?
> The model trained by u runs in the speed of about 100ms/frame on my PC.
> I trained a new model with your project on AISegment dataset.
> But this new model runs in the speed of about 450ms/frame.
> The accuracies are almost same but not for speed.
> What do u think the reason?
> Thanks

I have no idea of this issue, but may be the pre-trained model was trained by diffrent source code version or diffrent pytorch environment which improved the performance.
I did an experiment:

1、Load model directly from pre-trained 'model_obj.pth', the test result is fine.
2、Build model from current repo's source and load_state_dict from pre-trained 'ckpt_lastest.pth', the test result is quite worse.

So I think the source code is diffrent from the pre-trained model. This may be affect the performance too.

By upgrading pytorch version to 1.5-cuda10.1-cudnn7-devel, the performance  is much better on cpu. 
`docker pull pytorch/pytorch:1.5-cuda10.1-cudnn7-devel`

**Comment by rose-jinyang** [comment]
Thank you for your nice advice.
